lib/winapi/profiling.c

```diff
@@ -10,28 +10,19 @@
 #include <stddef.h>
 #include <xboxkrnl/xboxkrnl.h>
 
-BOOL QueryPerformanceCounter (LARGE_INTEGER *lpPerformanceCount)
-{
-    assert(lpPerformanceCount != NULL);
-
-    lpPerformanceCount->QuadPart = __rdtsc();
-    return TRUE;
-}
-
-BOOL QueryPerformanceFrequency (LARGE_INTEGER *lpFrequency)
-{
-    assert(lpFrequency != NULL);
-
 #ifdef USE_RDTSC_FOR_FREQ
-    #define AVG_SET 10
-    ULARGE_INTEGER f_rdtsc, avg;
-    ULONG f_ticks = 0;
+static LARGE_INTEGER frequency = {0, 0};
+static void __attribute__((constructor)) PrimeQueryPerformanceFrequency ()
+{
+    #define AVG_SET 2
+    ULARGE_INTEGER f_rdtsc, avg = {0, 0}, s_rdtsc;
+    ULONG f_ticks = 0, s_ticks = 0;
 
-    avg.QuadPart = 0;
+    Sleep(500);
```

Comment:
Why?

Reply:
Values generated soon after the system is booted seem to give less precise results, as to why; I'm not sure yet. Runs with a 1s initial wait and 500ms loop wait give us `733332000`, and calling it again even moments later bring us to `733333000`.

Reply:
I'd say that's good enough? What kind of precision do you expect? With your multiplication by 1000 at the end, this means that there's a single tick difference (1.36425648 nanoseconds).

That's a 0.00013636369% error.. how is that not negligible?

What are you using this function for that you require such precision?

Reply:
> What are you using this function for that you require such precision?

I don't need such precision. It's an attempt to placate you and other maintainers. I don't want a hard value, *you* do. So I've made it so it's as close as I think is reasonable (possible?).

Reply:
I'd say a sleep of less than a millisecond would be good in constructors. Maybe 10ms if necessary.
There's valid cases of sleeping in the network, USB and GPU driver - even up to 100ms wouldn't be too problematic (especially async).

But in this instance, there's a perfect solution which has no wait.
It outperforms your solution in every single way on a stock Xbox.

The only time it doesn't work is with CPU mods / under- or overclocking.
But that's not a case nxdk should support unless it's supportable with reasonable effort.
Hint: it likely isn't, because it becomes a problem of defining the limit of what is considered an Xbox. If I overclock the audio chip but that's used in some nxdk code to calibrate the overclocked CPU clock, then how could this ever work without telling nxdk what you did to your system?
I can see perfect reasons for overclocking either of them, but absolutely no way how to reasonably support it in nxdk.

We could support the more common case of overclocking the CPU, but not if it has major drawbacks for most users which run a stock Xbox (which is the majority).